src/controllers/user.controller.js

```diff
@@ -50,4 +55,74 @@ export default class UserController {
       });
     }
   }
+
+  async googleLogin(req, res) {
+    try {
+      const profile = req.user;
+
+      let user = await new UserService().getUser(profile.emails[0].value);
+
+      if (!user) {
+        user = await new UserService().createUser({
+          email: profile.emails && profile.emails[0].value,
+          password: null,
+          role_id: 4,
+          first_name: profile.name && profile.name.familyName,
+          last_name:
+            profile.name &&
+            [profile.name.middleName, profile.name.givenName].join(' '),
+          profile_picture: profile.photos && profile.photos[0].value,
+          provider: 'GOOGLE'
+        });
+      }
+
+      const token = generateToken({ email: user.email, id: user.id }, '7d');
+
+      return res.status(200).json({
+        status: 200,
+        message: 'Logged in successfully',
+        token
+      });
+    } catch (error) {
+      return res.status(500).json({
+        message: 'Error occured while logging in',
+        data: error.message
+      });
+    }
+  }
+
+  async facebookLogin(req, res) {
+    try {
+      const profile = req.user;
+
+      let user = await new UserService().getUser(profile.emails[0].value);
+
+      if (!user) {
+        user = await new UserService().createUser({
+          email: profile.emails && profile.emails[0].value,
+          password: null,
+          role_id: 4,
```

Comment:
Remove role

Reply:
Actually, I found out that I did use the relationship and as the role_id was the foreign key it can't be set as a default so it needs to be sent as 4 as the requester